wordpress-mobile/WordPress-Android
# Stories library from s3 as binary dependency

This PR removes the `stories-android` git submodule and instead adds it as a binary dependency. It depends on the following PRs:

* https://github.com/Automattic/stories-android/pull/713 adds support for publishing `stories-android` artifacts to S3. It needs to be merged first, so the version that's used in this PR can be updated.
* https://github.com/wordpress-mobile/WordPress-Android/pull/15368 adds localization support for the `stories-android` binary dependency. **This also fixes the CI error, so please ignore it for this PR.**
* https://github.com/wordpress-mobile/WordPress-Android/pull/15369 cleans up the CircleCI steps from the submodule commands.

It also adds support for composite build for `stories-android` which works exactly the same way as the other libraries.

_Note that I am leaving this PR as a draft, because it depends on other PRs, but I'd appreciate a review so that it can all be merged at once. Thanks!_

**To test:**
* Smoke test the stories publishing flow
* Smoke test the stories publishing flow by adding stories-android as a composite build. You can find the instructions for this in the `local-builds.gradle-example` file:

1. `cp local-builds.gradle-example local-builds.gradle`
2. Remove the `//` prefix for `localStoriesAndroidPath` and make sure the relevant path is correct
3. Go into `stories-android` folder and make sure `publish-to-s3` branch is checked out
4. Re-sync Android Studio and test the app.

## Regression Notes
1. Potential unintended areas of impact
Stories-android related features might stop working

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing the stories-android feature

3. What automated tests I added (or what prevented me from doing so)
N/A

**PR submission checklist:**
